scripts/test/utils

```diff
@@ -1,27 +1,54 @@
 #!/usr/bin/env bash
 
-set -e
+set -e -o pipefail
+
+### Constants ###
+
+readonly TEST_DIR=projects/releases
 
 ### General Functions ###
 
 function exit_error() {
-    echo "--------- Captured output ---------"
-    echo "${output}"
-    echo "-----------------------------------"
     echo "ERROR: $*"
     exit 1
 }
 
-function print_test() {
-    echo "TEST: $*"
+function running_test() {
+    [[ -z "$CI" ]] || { printf '::endgroup::\n::group::%s\n' "$*" && return 0; }
+    local div=${*//?/=}
+    printf '%s\n%s\n%s\n' "$div" "$*" "$div"
+}
+
+# Copy over current git repo to testing directory to test any changes and avoid affecting the local repo.
+function prepare_test_repo() {
+    rm -rf "$TEST_DIR"
+    mkdir -p "$TEST_DIR"
+    GLOBIGNORE=$(paste -sd: .gitignore)
+    cp -r -- * "$TEST_DIR"
```

Comment:
`.gitignore` patterns aren’t globs, so this won’t work in all cases.

If the test directory is supposed to be a git repository, it’s better to clone it:

    git clone . "$TEST_DIR"

Otherwise, archive it:

    git archive HEAD | tar -x -C "$TEST_DIR"